### PR TITLE
Add Moshi JSON Converter

### PR DESCRIPTION
**Pull request by S2606**
Add moshi converter for reducing boilerplate code

Fixes https://github.com/fossasia/susi_android/issues/848

### CODE REVIEW

**Comment by S2606** [comment]
@chiragw15 Please review

File: app/src/main/java/org/fossasia/susi/ai/rest/responses/others/WebIcon.kt [hunk 1]
```diff
@@ -1,19 +1,15 @@
...
  * Created by mayank on 12-12-2016.
  */
 
-class WebIcon {
+class WebIcon (
+
 
-@SerializedName("URL")
- @Expose
  var url: String? = null
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/rest/responses/others/WebIcon.kt]: @S2606 this is a single variable in whole file. Can we like add three related classes like webIcon, RelatedTopics, WebSearch in same file?

**Comment by betterclever** [comment]
An even better way to store classes for represent simple data items, say Java Beans, in Kotlin is **data class** https://kotlinlang.org/docs/reference/data-classes.html

But as far as I see, current Moshi version has a problem with their usage. https://github.com/square/moshi/issues/309

See if it could be used later.

**Comment by S2606** [comment]
@chiragw15 should i implement data class or stick with moshi??

**Comment by chiragw15** [comment]
@S2606 stick with moshi for now.

**Comment by betterclever** [comment]
@S2606 data classes are just for a better representation for data. It has nothing to do with Moshi. It is a recommended practice in Kotlin to use data classes for Beans. Though currently, the library has some problems with Moshi but you may want to implement later.

**Comment by S2606** [comment]
Thank you @betterclever for clearing it.

File: app/src/main/java/org/fossasia/susi/ai/rest/responses/others/WebIcon.kt (hunk 1 above)
Comment (chiragw15) [reply app/src/main/java/org/fossasia/susi/ai/rest/responses/others/WebIcon.kt]: @S2606 ^

Comment (S2606) [reply app/src/main/java/org/fossasia/susi/ai/rest/responses/others/WebIcon.kt]: @chiragw15 I am working on it. Sorry for delay as i had exams

Comment (S2606) [reply app/src/main/java/org/fossasia/susi/ai/rest/responses/others/WebIcon.kt]: @chiragw15 should i keep the file name as relatedtopic or change it??

Comment (chiragw15) [reply app/src/main/java/org/fossasia/susi/ai/rest/responses/others/WebIcon.kt]: Keep file name anything but keep class names same.

Comment (S2606) [reply app/src/main/java/org/fossasia/susi/ai/rest/responses/others/WebIcon.kt]: @chiragw15 The thing is on adding the other class it is showing an error

Comment (chiragw15) [reply app/src/main/java/org/fossasia/susi/ai/rest/responses/others/WebIcon.kt]: @S2606 take help from these https://android.jlelse.eu/keddit-part-6-api-retrofit-kotlin-d309074af0  and
https://github.com/juanchosaravia/KedditBySteps/commit/8295dfa18896a3ac18757e4c026fa1e7d5ec1600#diff-f419ad5f6c76e2617a08405003a66c76

**Comment by S2606** [comment]
@chiragw15 Please review

File: app/src/main/java/org/fossasia/susi/ai/rest/responses/others/RelatedWebTopics.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+class WebIcon ( var url: String? = null )
+
+class WebSearch (
+
+        var heading: String? = null,
+        var relatedTopics: List<RelatedTopics>? = null
+
+)
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/rest/responses/others/RelatedWebTopics.kt]: @S2606 Nice work here. Similarly, make 2 more files: one containing response from location. Other containing responses from susi. So, we will finally have one file corresponding to one client i.e Location client, SUSI, client and websearch client.

File: app/build.gradle [hunk 3]
```diff
@@ -76,6 +76,9 @@ dependencies {
     compile 'com.squareup.retrofit2:converter-gson:2.1.0'
```
Comment (chiragw15) [inline app/build.gradle]: If we are using moshi, we can remove gson.

**Comment by mariobehling** [comment]
@S2606 Thanks for your contribution. Please finalize this PR and resolve conflicts.

File: app/build.gradle (hunk 3 above)
Comment (S2606) [reply app/build.gradle]: @chiragw15 Is it ok if i resolve conflict from github interface or should i do it from git?

Comment (chiragw15) [reply app/build.gradle]: Anyone is fine.

**Comment by S2606** [comment]
@chiragw15 Due to some issues, I am closing this PR and will be sending a new one.

**Pull request closed without merging**